From 0fab9ef745e8450e841999ad1a7c2c23d2961a96 Mon Sep 17 00:00:00 2001
From: Andy Duan <fugang.duan@nxp.com>
Date: Thu, 26 Jul 2018 18:48:03 +0800
Subject: [PATCH 4282/5242] MLK-19048: pinctrl: add defer probe check for
 pinctrl setting assertion

commit  12b2568f689614196776c622d9304fa679c33e66 from
https://source.codeaurora.org/external/imx/linux-imx.git

Pinctrl support of a device tree property "pinctrl-assert-gpios"
under client device node to select function at a board level pin
multiplexer. The pin route is controlled by a GPIO or i2c/spi expander
GPIO.

For i2c/spi expander GPIO, it may be loaded after client device that
set "pinctrl-assert-gpios" property in devicetree. Then the client device's
pin function doesn't work.

So it should add defer probe check for the GPIO pin.

Acked-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pinctrl/devicetree.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/pinctrl/devicetree.c b/drivers/pinctrl/devicetree.c
index 62fccb8..1be9527 100644
--- a/drivers/pinctrl/devicetree.c
+++ b/drivers/pinctrl/devicetree.c
@@ -201,8 +201,11 @@ static int dt_gpio_assert_pinctrl(struct pinctrl *p)
 	for (;; index++) {
 		gpio = of_get_named_gpio_flags(np, "pinctrl-assert-gpios",
 					       index, &flags);
-		if (gpio < 0)
+		if (gpio < 0) {
+			if (gpio == -EPROBE_DEFER)
+				return gpio;
 			break; /* End of the phandle list */
+		}
 
 		if (!gpio_is_valid(gpio))
 			return -EINVAL;
-- 
1.7.9.5

